commit e0de2fe7651fa6875fbea739b3c6600842aa681f
Author: Manoj Gupta <manojgupta@google.com>
Date:   Tue Sep 12 20:14:37 2017 -0700

    [ARM] builtins: Do not abort in clear_cache.
    
    Summary:
    __builtion___clear_cache maps to clear_cache function which may call
    abort(). clear_cache documentation in libgcc does not specify any aborting
    behavior. And Libgcc implementation does not do abort for clear_cache.
    Fix by removing the abort.
    
    Fixes PR34588.
    
    Reviewers: rengolin, compnerd, srhines, peter.smith
    
    Subscribers: aemerson, kristof.beyls, llvm-commits
    
    Differential Revision: https://reviews.llvm.org/D37788

diff --git a/lib/builtins/clear_cache.c b/lib/builtins/clear_cache.c
index e21ac08f8..f67ab5d3f 100644
--- a/lib/builtins/clear_cache.c
+++ b/lib/builtins/clear_cache.c
@@ -121,9 +121,6 @@ void __clear_cache(void *start, void *end) {
                           : "=r"(start_reg)
                           : "r"(syscall_nr), "r"(start_reg), "r"(end_reg),
                             "r"(flags));
-         if (start_reg != 0) {
-             compilerrt_abort();
-         }
     #elif defined(_WIN32)
         FlushInstructionCache(GetCurrentProcess(), start, end - start);
     #else
